python/pyarrow/tests/test_extension_type.py

```diff
@@ -1130,13 +1131,32 @@ def test_cpp_extension_in_python(tmpdir):
 
 
 def test_tensor_type():
-    tensor_type = pa.FixedShapeTensorType(pa.int8(), (2, 3))
+    tensor_type = pa.fixed_shape_tensor(pa.int8(), [2, 3])
     assert tensor_type.extension_name == "arrow.fixed_shape_tensor"
     assert tensor_type.storage_type == pa.list_(pa.int8(), 6)
+    assert tensor_type.shape == [2, 3]
+    assert not tensor_type.dim_names
+    assert not tensor_type.permutation
```

Comment:
What is the exact return value? (I assume not False, but None or an empty list?) 
If it's an empty list, we should maybe check in the bindings for the attribute whether the underlying C++ vector is empty, and in that case return None instead? For a python API, that feels nicer to indicate "not defined", and also ensures that _if_ it returns a list, it is guaranteed to return a list of the same length as `shape`

Reply:
I think it returns `None`, will check.

Reply:
I checked and you were of course correct to assume an empty list =)
Will correct.